### PR TITLE
Feat: modify create/edit project forms - separate out enterprise settings UI

### PR DESCRIPTION
**Pull request by andreas-unleash**
Separates Enterprise Settings from ProjectForm
Separates Create and Edit project functionality based on pro or enterprise

Closes: # [1-1418](https://linear.app/unleash/issue/1-1418/frontend-move-project-settings-to-new-endpoint)

### CODE REVIEW

File: frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/CollaborationModeTooltip.tsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { Box, styled, Typography } from '@mui/material';
 import { FC } from 'react';
-import { HelpIcon } from 'component/common/HelpIcon/HelpIcon';
-import { useUiFlag } from 'hooks/useUiFlag';
-import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { HelpIcon } from '../../../common/HelpIcon/HelpIcon';
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/CollaborationModeTooltip.tsx:3]: Shorten

File: frontend/src/component/project/Project/CreateProject/CreateProject.tsx [hunk 2]
```diff
@@ -111,18 +103,10 @@ const CreateProject = () => {
                 projectId={projectId}
                 setProjectId={setProjectId}
                 projectName={projectName}
-                projectMode={projectMode}
```
Comment (sjaanus) [inline frontend/src/component/project/Project/CreateProject/CreateProject.tsx:114]: For creation, we still want project mode. Is this removing project mode from creation?

File: frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/CollaborationModeTooltip.tsx [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+import { useUiFlag } from '../../../../hooks/useUiFlag';
+import { ConditionallyRender } from '../../../common/ConditionallyRender/ConditionallyRender';
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/CollaborationModeTooltip.tsx:5]: Shorten

File: frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/FeatureFlagNamingTooltip.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { Box } from '@mui/material';
 import { FC } from 'react';
-import { HelpIcon } from 'component/common/HelpIcon/HelpIcon';
+import { HelpIcon } from '../../../common/HelpIcon/HelpIcon';
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/FeatureFlagNamingTooltip.tsx:3]: Shorten

File: frontend/src/component/project/Project/CreateProject/CreateProject.tsx [hunk 5]
```diff
@@ -111,18 +103,10 @@ const CreateProject = () => {
...
                 projectStickiness={projectStickiness}
-                featureLimit={featureLimit}
-                featureNamingExample={featureNamingExample}
-                featureNamingPattern={featureNamingPattern}
-                setFeatureNamingPattern={setFeatureNamingPattern}
-                featureNamingDescription={featureNamingDescription}
-                setFeatureNamingDescription={setFeatureNamingDescription}
-                setFeatureNamingExample={setFeatureNamingExample}
+                featureLimit={''}
```
Comment (sjaanus) [inline frontend/src/component/project/Project/CreateProject/CreateProject.tsx:107]: Feature limit should not be on creation

File: frontend/src/component/project/Project/CreateProject/CreateProject.tsx (hunk 2 above)
Comment (sjaanus) [reply frontend/src/component/project/Project/CreateProject/CreateProject.tsx:114]: Furthermore, it seems to lack defaul value, which should be open

image

Comment (sjaanus) [reply frontend/src/component/project/Project/CreateProject/CreateProject.tsx:114]: Also, the dropdown seems to be unselectable.

File: frontend/src/component/project/Project/ProjectForm/ProjectForm.tsx [hunk 6]
```diff
@@ -1,38 +1,30 @@
...
 import Input from 'component/common/Input/Input';
 import { FeatureTogglesLimitTooltip } from './FeatureTogglesLimitTooltip';
-import { FeatureFlagNamingTooltip } from './FeatureFlagNamingTooltip';
-import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
-import { usePlausibleTracker } from 'hooks/usePlausibleTracker';
-import { useUiFlag } from 'hooks/useUiFlag';
+import { ProjectMode } from '../hooks/useProjectEnterpriseSettingsForm';
+import useUiConfig from '../../../../hooks/api/getters/useUiConfig/useUiConfig';
+import { CollaborationModeTooltip } from '../ProjectEnterpriseSettingsForm/CollaborationModeTooltip';
+import Select from '../../../common/select';
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectForm/ProjectForm.tsx:11]: Shorten

File: frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/EditProject.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
+import { UPDATE_PROJECT } from '../../../../../providers/AccessProvider/permissions';
+import useProject from '../../../../../../hooks/api/getters/useProject/useProject';
+import useUiConfig from '../../../../../../hooks/api/getters/useUiConfig/useUiConfig';
+import { useRequiredPathParam } from '../../../../../../hooks/useRequiredPathParam';
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/EditProject.tsx:4]: Shorten

**Comment by sjaanus** [comment]
Can we have less space on bottom of these sections, because we have 3 of sections and it feels alot.

image

File: frontend/src/component/project/Project/ProjectForm/ProjectForm.tsx [hunk 8]
```diff
@@ -94,95 +81,28 @@ const StyledInputContainer = styled('div')(() => ({
...
-        if (!matches) {
-            return { state: 'invalid', reason: 'Example does not match regex' };
-        } else {
-            return { state: 'valid' };
-        }
-    }
-    return { state: 'valid' };
-};
-
-const useFeatureNamePatternTracking = () => {
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectForm/ProjectForm.tsx:132]: This removes plausible tracking from project creation.

**Review by sjaanus — CHANGES_REQUESTED** [review]

File: frontend/src/component/project/Project/ProjectForm/ProjectForm.tsx (hunk 8 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectForm/ProjectForm.tsx:132]: FeatureNaming is not part of project creation

Comment (sjaanus) [reply frontend/src/component/project/Project/ProjectForm/ProjectForm.tsx:132]: Yes, you are correct.

File: frontend/src/component/project/Project/CreateProject/CreateProject.tsx [hunk 9]
```diff
@@ -30,25 +30,18 @@ const CreateProject = () => {
...
         setProjectId,
         setProjectName,
         setProjectDesc,
-        getProjectPayload,
+        getCreateProjectPayload,
         clearErrors,
         validateProjectId,
         validateName,
         setProjectStickiness,
         setFeatureLimit,
```
Comment (sjaanus) [inline frontend/src/component/project/Project/CreateProject/CreateProject.tsx:44]: Can be removed.

File: frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/ProjectEnterpriseSettingsForm.tsx [hunk 10]
```diff
@@ -0,0 +1,367 @@
+import React, { useEffect } from 'react';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import Select from 'component/common/select';
+import { ProjectMode } from '../hooks/useProjectEnterpriseSettingsForm';
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/ProjectEnterpriseSettingsForm.tsx:4]: Can be shortened

File: frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/DeleteProjectForm.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { DeleteProject } from '../DeleteProject';
+import FormTemplate from 'component/common/FormTemplate/FormTemplate';
+import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
+import useProjectApi from 'hooks/api/actions/useProjectApi/useProjectApi';
+import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
+import { PageContent } from '../../../../../common/PageContent/PageContent';
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/DeleteProjectForm.tsx:7]: Can be shortened

File: frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/UpdateEnterpriseSettings.tsx [hunk 12]
```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
+import useToast from 'hooks/useToast';
+import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
+import useProjectEnterpriseSettingsForm from '../../../hooks/useProjectEnterpriseSettingsForm';
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/UpdateEnterpriseSettings.tsx:5]: Shorten

File: frontend/src/hooks/api/actions/useProjectApi/useProjectApi.ts [hunk 13]
```diff
@@ -68,6 +82,25 @@ const useProjectApi = () => {
         }
     };
 
+    const editProjectSettings = async (
```
Comment (sjaanus) [inline frontend/src/hooks/api/actions/useProjectApi/useProjectApi.ts:85]: We should use orval types here and throughout this logic.

**Review by sjaanus — CHANGES_REQUESTED** [review]

File: frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/EditProject.tsx [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+        </Alert>
+    );
+
+    return (
+        <>
+            {accessDeniedAlert}
+            <StyledFormContainer>
+                <UpdateProject project={project} />
+                <ConditionallyRender
+                    condition={isEnterprise()}
```
Comment (sjaanus) [inline frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/EditProject.tsx:37]: Can we add a test for this?

**Review by sjaanus — CHANGES_REQUESTED** [review]

File: frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/ProjectEnterpriseSettingsForm.tsx (hunk 10 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/ProjectEnterpriseSettingsForm.tsx:4]: This is actually a hook in the parent folder

File: frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/UpdateEnterpriseSettings.tsx (hunk 12 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectSettings/Settings/EditProject/UpdateEnterpriseSettings.tsx:5]: not shortened in this case but aliased

File: frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/ProjectEnterpriseSettingsForm.tsx (hunk 10 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectEnterpriseSettingsForm/ProjectEnterpriseSettingsForm.tsx:4]: but can be aliased if the dots are the problem

File: frontend/src/hooks/api/actions/useProjectApi/useProjectApi.ts (hunk 13 above)
Comment (andreas-unleash) [reply frontend/src/hooks/api/actions/useProjectApi/useProjectApi.ts:68]: Done but with problems with the generation

**Comment by andreas-unleash** [comment]
Closing for cleanup after biome merge

**Pull request closed without merging**